src/auth/src/credentials/user_credential.rs

```diff
@@ -26,7 +26,14 @@ use time::OffsetDateTime;
 const OAUTH2_ENDPOINT: &str = "https://oauth2.googleapis.com/token";
 
 pub(crate) fn creds_from(js: serde_json::Value) -> Result<Credential> {
-    let token_provider = UserTokenProvider::from_json(js)?;
+    let au = serde_json::from_value::<AuthorizedUser>(js)
+        .map_err(|e| CredentialError::new(false, e.into()))?;
```

Comment:
drive-by: I think if if you had `CredentialError::non_retryable<T: Into<...>>(e: T)` you could write:

```rust
.map_err(CredentialError::non_retryable)
```

You could make it `pub(crate)` if you don't want to grow your public API.

Reply:
Yeah, that is much nicer. I tried this once when I was 3-weeks worse at Rust. Now I can make it work.